sql/builtin_functions.cc

```diff
@@ -1905,6 +1905,39 @@ BoundFunction horizontal_max(const std::vector<BoundSqlExpression> & args)
 }
 static RegisterBuiltin registerHorizontal_Max(horizontal_max, "horizontal_max");
 
+BoundFunction horizontal_max_key(const std::vector<BoundSqlExpression> & args)
+{
+    return {[=] (const std::vector<ExpressionValue> & args,
+                 const SqlRowScope & context) -> ExpressionValue
+            {
+                double max_val = nan("");
+                Utf8String max_key;
+                Date ts = Date::negativeInfinity();
+
+                auto onAtom = [&] (const Coord & columnName,
+                                   const Coord & prefix,
+                                   const CellValue & val,
+                                   Date atomTs)
+                    {
+                        if (!val.empty()) {
+                            double curr = val.toDouble();
+                            if(std::isnan(max_val) || curr > max_val) {
+                                ts = atomTs;
+                                max_val = curr;
+                                max_key = prefix.toString() + "." + columnName.toString();
```

Comment:
May be we can capture the prefix and column name separately and perform the string operation outside the loop.